### PR TITLE
[feat] refactor the CaseDetails component

### PR DESCRIPTION
**Pull request by ccatherinetan**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### :palm_tree: What's new in this PR
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
* refactors CaseDetails component to take in any Listing rather than only CaseListing → renamed to ListingDetails 
* updates frontend to match Kyrene's designs 

### :evergreen_tree: Screenshots
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @kyrenetam at the bottom."

Case 
Screen Shot 2024-04-08 at 3 39 42 PM

Case Interpretation
Screen Shot 2024-04-08 at 3 39 21 PM

LCA  (research topics icon has been updated, but not included in the screenshot)
Screen Shot 2024-04-08 at 3 38 40 PM

Doc Translation
Screen Shot 2024-04-08 at 3 36 51 PM

One-Time Interpretation
Screen Shot 2024-04-08 at 3 38 57 PM

## :deciduous_tree: How to review
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
* checkout this branch, `npm run dev`, nav to http://localhost:3000/cases 
* (i might try to create a test page to see if it works on Language Support and LCA's later)
* play around with `app/listing-details-test/page` to see how different listing types are rendered (see http://localhost:3000/listing-details-test)

## :seedling: Next steps 
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
* refactor interest form to take Listing rather than CaseListing
* use ListingDetails component on LCA and Lang Support pages (test that the non-case listing types work correctly)
* finalize nitty-gritty of front-end: style system

### :information_source: Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### :potted_plant: Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @varortz
@kyrenetam

### CODE REVIEW

File: src/components/ListingDetails/index.tsx [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+const intepretationFields = [
+  // Languages
+  {
+    label: 'Language(s)',
+    getValue: (data: Listing) => (data as Interpretation).languages.join(', '),
+  },
+  // Interpretation Type
+  {
+    label: 'Interpretation Type', // 'Language Support Type' for consistency w/ Doc?
```
Comment (varortz) [inline src/components/ListingDetails/index.tsx]: Could we change the label to "Language Support Type" so that the naming is consistent across all listings on the Language Support page?

File: src/components/ListingDetails/index.tsx [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+  },
+  // Custody Location
+  {
+    // Temporary
+    label: 'Custody Location',
+    getValue: () => 'San Diego, CA',
+  },
+];
+
+const caseInterpretationFields = [
```
Comment (varortz) [inline src/components/ListingDetails/index.tsx]: I think we're missing the "Language Support Type" field here, could we go ahead and add it to stay consistent with other listings on the Language Support page?

File: src/components/ListingDetails/index.tsx [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+  // Languages
+  {
+    label: 'Language(s)',
+    getValue: (data: Listing) =>
+      (data as DocumentTranslation).languages.join(', '),
+  },
+  // Language Support Type
+  {
+    label: 'Language Support Type',
+    getValue: (data: Listing) => 'Written',
```
Comment (varortz) [inline src/components/ListingDetails/index.tsx]: thoughts on changing the value to "Document Translation"?

File: src/components/ListingDetails/index.tsx [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+  },
+  // Client Location
+  {
+    label: 'Client Location',
+    getValue: (data: Listing) => (data as CaseListing).client_location || 'N/A',
+  },
+  // Custody Location
+  {
+    // Temporary
+    label: 'Custody Location',
```
Comment (varortz) [inline src/components/ListingDetails/index.tsx]: could we go ahead and remove this field? I think we should either be showing a "Client Location" or "Custody Location" label but not both, depending on whether the case's `is_detained` field is false/true.

File: src/components/ListingDetails/index.tsx [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+];
+
+export default function ListingDetails({
+  listingData,
+}: {
+  listingData: Listing;
+}) {
+  const listingFields = () => {
+    if (listingData.listing_type === 'CASE') {
+      if (listingData.needs_interpreter) {
```
Comment (varortz) [inline src/components/ListingDetails/index.tsx]: I think we may want a different if-check for case interpretations. Based off of this implementation, we'd always be returning `caseInterpretationFields` when `listingData.needs_interpreter == true` even if we intend to return `caseFields`. We should instead use an optional `interpretation` prop to help indicate that we're dealing with case interpretations. The `ListingCard` component uses this approach, so it might be helpful to take a quick look at it for reference!

Comment (varortz) [reply src/components/ListingDetails/index.tsx]: I think this should also answer the question you left in the comment below. **Quick recap:** We should return `caseFields` for the cases page and `caseInterpretationFields` for the Language Support page, and we could use an optional `interpretation` prop to help distinguish which page we're on.

File: src/components/ListingDetails/index.tsx [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+const intepretationFields = [
+  // Languages
+  {
+    label: 'Language(s)',
+    getValue: (data: Listing) => (data as Interpretation).languages.join(', '),
+  },
+  // Interpretation Type
+  {
+    label: 'Interpretation Type', // 'Language Support Type' for consistency w/ Doc?
+    getValue: (data: Listing) => 'One-Time Intepretation',
```
Comment (varortz) [inline src/components/ListingDetails/index.tsx]: since we don't use `data` to determine the value of this field we can simplify this to:
 `getValue: () => 'One-Time Intepretation'`

File: src/components/ListingDetails/index.tsx (hunk 3 above)
Comment (varortz) [reply src/components/ListingDetails/index.tsx]: Also, we can simplify this to `getValue: () => 'Written'` because we aren't using `data` to determine the value of this field.

File: src/components/ListingDetails/styles.tsx [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+export const CaseDisplay = styled.aside`
+  position: relative;
+  width: 100%;
+`;
+
+export const CaseTitle = styled(H1)`
+  margin: 0.4rem 4rem;
+`;
+
+// export const CaseInterestContainer = styled.div`
```
Comment (varortz) [inline src/components/ListingDetails/styles.tsx]: let's go ahead and remove these commented lines if we won't need them moving forward

File: src/components/ListingDetails/styles.tsx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+  gap: 20px;
+  padding: 30px 55px;
+  border: 1px solid ${COLORS.greyLighter};
+`;
+
+export const InlineP = styled(P)`
+  display: inline;
+`;
+
+export const DateText = styled(P)<{ $bold?: true }>`
```
Comment (varortz) [inline src/components/ListingDetails/styles.tsx]: should we add these to `styles/text.tsx`?

File: src/components/ListingDetails/index.tsx [hunk 9]
```diff
@@ -0,0 +1,320 @@
...
+    if (listingData.listing_type === 'INT') {
+      return intepretationFields;
+    }
+    if (listingData.listing_type === 'DOC') {
+      return docFields;
+    }
+    return LCAFields;
+  }
+
+  const dateComponent = () => {
```
Comment (varortz) [inline src/components/ListingDetails/index.tsx]: looks like eslint is giving an error that we should probably fix: 
image

File: src/components/ListingDetails/index.tsx [hunk 10]
```diff
@@ -0,0 +1,320 @@
...
+    } 
+  };
+
+  const profile = useContext(ProfileContext);
+
+  const Interest = useMemo(() => {
+    if (profile && profile.userId)
+      return profile.profileData ? (
+        // <InterestForm caseData={listingData} />
+        <H3>Updated Interest Form Coming Soon!</H3> // remove after refactoring InterestForm
```
Comment (varortz) [inline src/components/ListingDetails/index.tsx]: quick little nit: can we show the interest form for the cases page? To do so, we can update this line to: 
`<InterestForm caseData={listingData as CaseListing} />`

We'd also need to update line 317 the following code:
`      <BorderedSection>
        {listingData.listing_type === 'CASE' ? (
          Interest
        ) : (
          <H3>Updated Interest Form Coming Soon!</H3>
        )}
      </BorderedSection>`

File: src/app/cases/styles.ts [hunk 11]
```diff
@@ -65,7 +65,7 @@ export const CardColumn = styled.div`
 
 export const CaseDetailsContainer = styled.div`
```
Comment (varortz) [inline src/app/cases/styles.ts:66]: for the `ListingCard` column, I believe @jinkang-0  added styling for a custom scrollbar. Do we wanna add it here as well (this ensures the scrollbars are always consistent regardless of the device)?? @kyrenetam curious what you think of the look of this as well! Here's a screenshot:

image

File: src/components/ListingDetails/index.tsx [hunk 12]
```diff
@@ -0,0 +1,320 @@
...
+                ? listingData.research_topic
+                : 'Not Available'}
+            </P>
+          </BorderedSection>
+        )}
+        <BorderedSection>
+          <IconTextGroup>
+            <Icon type="briefcase" />
+            <H4>
+              {listingData.listing_type === 'LCA'
```
Comment (varortz) [inline src/components/ListingDetails/index.tsx]: Noticing that One-Time Transalations & Document Translations have case-specific labels ("Case Description" and "Case Highlights"), and I think this could get confusing for the user. Could we update this label as well as the Highlights label so that:
-  for One-Time Interpretations the labels are "Interpretation Description" and "Interpretation Highlights"
- for Document Translations the labels are "Assignment Description" and "Assignment Highlights"

File: src/components/ListingDetails/styles.tsx [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+//   display: block;
+//   overflow-y: auto;
+// `;
+
+export const InfoContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  min-height: 50%;
+  /* gap: 1.35rem;
```
Comment (varortz) [inline src/components/ListingDetails/styles.tsx]: can we remove these commented out lines?

File: src/components/ListingDetails/index.tsx [hunk 14]
```diff
@@ -0,0 +1,320 @@
...
+import Icon from '../Icon';
+import {
+  CaseDisplay,
+  InfoContainer,
+  FieldContainer,
+  InnerFieldContainer,
+  AuthButtons,
+  IconTextGroup,
+  BorderedSection,
+  // InlineP
```
Comment (varortz) [inline src/components/ListingDetails/index.tsx]: can we remove this if we're not using it within the file?

File: src/components/ListingDetails/styles.tsx (hunk 8 above)
Comment (varortz) [reply src/components/ListingDetails/styles.tsx]: actually noticed that we're not using `inlineP`, should we go ahead and remove it?

File: src/components/ListingDetails/index.tsx [hunk 15]
```diff
@@ -0,0 +1,320 @@
+/* eslint-disable react/jsx-no-comment-textnodes */
+import React, { useContext, useMemo } from 'react';
+import {
+  // timestampStringToDate,
+  // parseDate,
```
Comment (varortz) [inline src/components/ListingDetails/index.tsx]: can we remove these commented out lines?

File: src/app/listing-details-test/page.tsx [hunk 16]
```diff
@@ -0,0 +1,77 @@
+'use client';
```
Comment (varortz) [inline src/app/listing-details-test/page.tsx]: TY FOR MAKING A TESTING PAGE <3

File: src/components/ListingDetails/index.tsx (hunk 3 above)
Comment (ccatherinetan) [reply src/components/ListingDetails/index.tsx]: I'm down to change it to "Document Translation"! The figma has "Written," but i've left a comment on figma about this for @kyrenetam

File: src/components/ListingDetails/index.tsx (hunk 5 above)
Comment (ccatherinetan) [reply src/components/ListingDetails/index.tsx]: sg!

File: src/components/ListingDetails/styles.tsx (hunk 8 above)
Comment (ccatherinetan) [reply src/components/ListingDetails/styles.tsx]: sg! yes i removed InlineP!

File: src/components/ListingDetails/index.tsx (hunk 9 above)
Comment (ccatherinetan) [reply src/components/ListingDetails/index.tsx]: yes! fixed

File: src/components/ListingDetails/index.tsx (hunk 10 above)
Comment (ccatherinetan) [reply src/components/ListingDetails/index.tsx]: sg! added it back

File: src/components/ListingDetails/index.tsx [hunk 17]
```diff
@@ -0,0 +1,332 @@
...
+      </>
+    );
+  }, [listingData, profile]);
+
+  return (
+    <CaseDisplay>
+      <InfoContainer>
+        <BorderedSection>
+          {dateComponent()}
+          <Flex $align="center" $gap="18px" $direction="row">
```
Comment (jinkang-0) [inline src/components/ListingDetails/index.tsx]: nit: `Flex` defaults to `$direction='row'`, we can remove this prop

File: src/components/ListingDetails/styles.tsx [hunk 18]
```diff
@@ -61,3 +56,24 @@ export const AuthButtons = styled.div`
...
+  flex-direction: column;
+  gap: 20px;
+  padding: 30px 55px;
+  border: 1px solid ${COLORS.greyLighter};
+`;
+
+export const DateText = styled(P)<{ $bold?: true }>`
+  display: inline;
+  font-size: 1rem;
+  font-weight: ${props => (props.$bold ? 500 : 400)};
```
Comment (jinkang-0) [inline src/components/ListingDetails/styles.tsx]: let's use destructuring here

File: src/components/ListingDetails/index.tsx [hunk 19]
```diff
@@ -0,0 +1,332 @@
...
+    getValue: () => 'Document Translation', // check if this crashes
+  },
+  // Number of Pages
+  {
+    label: 'Number of Pages',
+    getValue: (data: Listing) => (data as DocumentTranslation).num_pages,
+  },
+];
+
+const LCAFields = [
```
Comment (jinkang-0) [inline src/components/ListingDetails/index.tsx]: nit: can we lowercase this? (i.e. `lcaFields`)

Comment (ccatherinetan) [reply src/components/ListingDetails/index.tsx]: yes!

File: src/components/ListingDetails/index.tsx (hunk 12 above)
Comment (ccatherinetan) [reply src/components/ListingDetails/index.tsx]: just to confirm, after last worksession, have we confirmed that we will standardize the naming across all listing types as "Description" and "Highlights" rather than "Case Highlights" or "Assignment Highlights"?

File: src/components/ListingDetails/index.tsx [hunk 20]
```diff
@@ -0,0 +1,323 @@
...
+  }, [listingData, profile]);
+
+  return (
+    <CaseDisplay>
+      <InfoContainer>
+        <BorderedSection>
+          {dateComponent()}
+          <Flex $align="center" $gap="18px" $direction="row">
+            <img src={logo.src} alt="DOHS Logo" width="77" height="77" />
+            <H2>{listingData.title || 'Migrant seeking representation'}</H2>
```
Comment (ccatherinetan) [inline src/components/ListingDetails/index.tsx]: small issue: I wasn't sure if we wanted to keep this default title for assignments. My understanding is that all the case titles default to "Migrant Seeking representation" if they are untitled on LegalServer. However, the title field for `cases` table is nullable on Supabase, so I'm wondering if there's ever a case where a title field would be entry; in that case, I will leave this here.

File: src/components/ListingDetails/index.tsx [hunk 21]
```diff
@@ -0,0 +1,323 @@
...
+    );
+  }, [listingData, profile]);
+
+  return (
+    <CaseDisplay>
+      <InfoContainer>
+        <BorderedSection>
+          {dateComponent()}
+          <Flex $align="center" $gap="18px" $direction="row">
+            <img src={logo.src} alt="DOHS Logo" width="77" height="77" />
```
Comment (jinkang-0) [inline src/components/ListingDetails/index.tsx]: let's remove this

File: src/components/ListingDetails/index.tsx (hunk 20 above)
Comment (jinkang-0) [inline src/components/ListingDetails/index.tsx]: can you refactor this to be different depending on the type of listing? (i.e. defaults for LCA and the various language support types)

we can ask Ambreen for the specifics on Monday, but feel free to add arbitrary defaults for now!

File: src/components/ListingDetails/index.tsx [hunk 22]
```diff
@@ -0,0 +1,323 @@
...
+            <H2>{listingData.title || 'Migrant seeking representation'}</H2>
+          </Flex>
+        </BorderedSection>
+        <BorderedSection>
+          <IconTextGroup>
+            <Icon type="tag" />
+            <H4>
+              {listingData.listing_type === 'LCA' ? 'Assignment' : 'Case'}{' '}
```
Comment (jinkang-0) [inline src/components/ListingDetails/index.tsx]: Let's remove this so it's just "Highlights"!

File: src/components/ListingDetails/index.tsx (hunk 12 above)
Comment (jinkang-0) [reply src/components/ListingDetails/index.tsx]: Confirmed: "Highlights" and "Description"

File: src/components/ListingDetails/index.tsx (hunk 20 above)
Comment (ccatherinetan) [reply src/components/ListingDetails/index.tsx]: i'll leave it as is for now, since all listing types besides cases have non-nullable titles, so hopefully the a default title will never be needed for non-case listings.

File: src/app/cases/styles.ts (hunk 11 above)
Comment (ccatherinetan) [reply src/app/cases/styles.ts:66]: ooo, during worksesh i asked @jinkang-0 if he could add it, but i'll coordinate this further

Comment (jinkang-0) [reply src/app/cases/styles.ts:66]: from worksesh: i add scroll

File: src/components/ListingDetails/index.tsx (hunk 20 above)
Comment (varortz) [reply src/components/ListingDetails/index.tsx]: Since admin upload LCA, Doc Translation and One-Time Interpretation requests, all these listings should technically not need a default title since we make it a required field on Retool upon submission

File: src/components/ListingDetails/index.tsx [hunk 23]
```diff
@@ -0,0 +1,424 @@
...
+  parseTimeCommitment,
+} from '@/utils/helpers';
+import { H2, H3, H4, P, StrongP } from '@/styles/text';
+import {
+  CaseListing,
+  DocumentTranslation,
+  Interpretation,
+  LimitedCaseAssignment,
+  Listing,
+} from '@/types/schema';import { useProfile } from '@/utils/ProfileProvider';
```
Comment (varortz) [inline src/components/ListingDetails/index.tsx]: could we put the second import on its own line?

File: src/components/ListingDetails/index.tsx [hunk 24]
```diff
@@ -0,0 +1,424 @@
...
+      let dateData = '';
+      if (listingData.listing_type === 'CASE') {
+        dateHeader = 'Next Court/Filing Date';
+        dateData = formatTimestamp(listingData.upcoming_date);
+      } else {
+        dateData = formatTimestamp(listingData.deadline);
+      }
+      return (
+        <IconTextGroup>
+          <Icon type="calendar" />
```
Comment (varortz) [inline src/components/ListingDetails/index.tsx]: I think this calendar icon is the one used in the `ListingCard` component so it looks a little smaller than expected. Could we add the calendar icon that's used in the `CaseDetails` on the Figma?

File: src/app/cases/styles.ts [hunk 25]
```diff
@@ -65,7 +65,7 @@ export const CardColumn = styled.div`
...
   overflow-y: scroll;
-  padding: 2.7rem 3rem 1.7rem 3rem;
+  /* padding: 2.7rem 3rem 1.7rem 3rem; */
```
Comment (jinkang-0) [inline src/app/cases/styles.ts]: nit: can we remove this comment

File: src/components/ListingDetails/index.tsx [hunk 26]
```diff
@@ -0,0 +1,410 @@
...
+        ];
+      } else {
+        fields = [
+          ...caseFields,
+          {
+            label: 'Client Location',
+            getValue: data => data.client_location || 'N/A',
+          },
+        ];
+      }
```
Comment (jinkang-0) [inline src/components/ListingDetails/index.tsx]: we can consolidate this into one else and a ternary:
i.e.
```typescript
if (interpretation && listingData.needs_interpreter) {
  ...
} else {
  fields = [
    ...caseFields,
    {
      label: (listingData.is_detained) ? 'Custody Location' : 'Client Location',
      ...
    }
  ]
}
```

File: src/components/ListingDetails/styles.tsx
Comment (jinkang-0) [inline src/components/ListingDetails/styles.tsx:1]: nit: can you rename this to `.ts` since we're not defining any JSX markup here?

File: src/components/ListingDetails/index.tsx [hunk 27]
```diff
@@ -0,0 +1,410 @@
...
+                : 'Not Available'}
+            </P>
+          </BorderedSection>
+        )}
+        <BorderedSection>
+          <IconTextGroup>
+            <Icon type="description" />
+            <H4>Description</H4>
+          </IconTextGroup>
+          <P>{listingData.summary ? listingData.summary : 'Not Available'}</P>
```
Comment (jinkang-0) [inline src/components/ListingDetails/index.tsx]: nit: we can shorten this to `listingData.summary || 'Not Available'`

File: src/components/ListingDetails/index.tsx [hunk 28]
```diff
@@ -0,0 +1,410 @@
...
+        {listingData.listing_type === 'LCA' && (
+          <BorderedSection>
+            <IconTextGroup>
+              <Icon type="research" />
+              <H4>Research Topic(s)</H4>
+            </IconTextGroup>
+            <P>
+              {listingData.research_topic
+                ? listingData.research_topic
+                : 'Not Available'}
```
Comment (jinkang-0) [inline src/components/ListingDetails/index.tsx]: nit: we can shorten this to `listingData.research_topic || 'Not Available'`

File: src/utils/helpers.ts [hunk 29]
```diff
@@ -94,13 +94,19 @@ export const parseTimeCommitment = (
...
+    return `${hoursPerWeek} hours/week`;
 
   const unit = numWeeks > 4 ? 'month' : 'week';
   const numUnit = numWeeks > 4 ? numWeeks / 4 : numWeeks;
   const plural = numUnit > 1 ? 's' : '';
-  if (hoursPerWeek === undefined) return `${numWeeks} ${unit}${plural}`;
+  if (hoursPerWeek === undefined || hoursPerWeek === null)
+    return `${numWeeks} ${unit}${plural}`;
 
   const rate = numWeeks > 4 ? hoursPerWeek / 4 : hoursPerWeek;
```
Comment (jinkang-0) [inline src/utils/helpers.ts]: this is mb: let's do `rate = numWeeks > 4 ? hoursPerWeek * 4 : hoursPerWeek` instead

Comment (varortz) [reply src/utils/helpers.ts]: oh yeah good catch!! I noticed this during Bush Bash as well but forgot to note it down

Comment (ccatherinetan) [reply src/utils/helpers.ts]: fixed!

File: src/components/ListingDetails/index.tsx [hunk 30]
```diff
@@ -0,0 +1,400 @@
...
+        dateHeader = 'Next Court/Filing Date';
+        dateData = formatTimestamp(listingData.upcoming_date);
+      } else {
+        dateData = formatTimestamp(listingData.deadline);
+      }
+      return (
+        <IconTextGroup>
+          <Icon type="calendarBig" />
+          <Subheading $bold>
+            {dateHeader}: <Subheading> {dateData}</Subheading>
```
Comment (varortz) [inline src/components/ListingDetails/index.tsx]: it seems like nesting `<p>` elements is giving us this console error:
image

Let's go ahead and move the nested `Subheading` outside so that it's at the same level as the other `Subheading` and `Icon`

Comment (ccatherinetan) [reply src/components/ListingDetails/index.tsx]: gotcha!

**Review by varortz — APPROVED** [review]
GREAT WORK CATHERINE!! :) Looks good to merge

File: src/components/ListingDetails/index.tsx [hunk 31]
```diff
@@ -0,0 +1,399 @@
...
+  interpretation = false,
+}: {
+  listingData: Listing;
+  interpretation?: boolean;
+}) {
+  const auth = useAuth();
+  const profile = useProfile();
+
+  const listingFields = useMemo(() => {
+    let fields: ListingField<any>[];
```
Comment (jinkang-0) [inline src/components/ListingDetails/index.tsx]: `any` should generally not be used unless it really can be any type
in this case, it shouldn't be `any`, but rather some type that extends `Listing` (i.e. `number` won't work)

this is a nitpick by Typescript (and more specifically, an [ESLint rule](https://typescript-eslint.io/rules/no-explicit-any/))

it is annoying, and i don't see an easy way to get around this except to simply return at every `if` check
to simplify this, you may want to implement a helper component like so:
```tsx
function ListingFields<T extends Listing>({
  fields,
  listingData,
}: {
  fields: ListingField<T>[];
  listingData: T;
}) {
  return (
    <InnerFieldContainer>
      {fields.map(({ label, getValue }) => (
        <div key={label}>{renderField(label, getValue(listingData))}</div>
      ))}
    </InnerFieldContainer>
  );
}
```

File: src/components/ListingDetails/index.tsx [hunk 32]
```diff
@@ -0,0 +1,399 @@
...
+    getValue: data => parseTimeCommitment(data.hours_per_week, data.num_weeks),
+  },
+  // Remote/In Person
+  {
+    label: 'Remote/In Person',
+    getValue: data => (data.is_remote ? 'Remote' : 'In Person'),
+  },
+];
+
+const intepretationFields: ListingField<Interpretation>[] = [
```
Comment (jinkang-0) [inline src/components/ListingDetails/index.tsx]: nit: should this be `interpretationFields`? (minor typo)

File: src/components/ListingDetails/styles.ts [hunk 33]
```diff
@@ -10,27 +10,11 @@ export const CaseDisplay = styled.aside`
 export const CaseTitle = styled(H1)`
   margin: 0.4rem 4rem;
 `;
```
Comment (jinkang-0) [inline src/components/ListingDetails/styles.ts]: while we're modifying this file, let's go ahead and switch to pixels as well!

File: src/components/ListingDetails/index.tsx (hunk 32 above)
Comment (ccatherinetan) [reply src/components/ListingDetails/index.tsx]: yes! good catch

File: src/components/ListingDetails/styles.ts (hunk 33 above)
Comment (ccatherinetan) [reply src/components/ListingDetails/styles.ts]: sg! i changed all the other gap/margins to px as well!

File: src/components/ListingDetails/index.tsx (hunk 31 above)
Comment (ccatherinetan) [reply src/components/ListingDetails/index.tsx]: sg! i've implemented ur suggestion!

File: src/components/ListingDetails/index.tsx [hunk 34]
```diff
@@ -0,0 +1,421 @@
...
+
+  const Interest = useMemo(() => {
+    if (auth && auth.userId) {
+      return profile?.profileData ? (
+        <InterestForm caseData={listingData as CaseListing} />
+      ) : (
+        <>
+          <H3>
+            Please finish submitting your profile before submitting interest.
+          </H3>
```
Comment (jinkang-0) [inline src/components/ListingDetails/index.tsx]: nit: not sure if it was ever updated since i couldn't find this particular design in the current iterations, but i believe the old text (david era) was "Please complete your profile before submitting interest"

image


I think that makes more sense semantically than submitting a profile..

File: src/components/ListingDetails/index.tsx [hunk 35]
```diff
@@ -0,0 +1,421 @@
...
+      return null;
+    }
+    return (
+      <Subheading $color={COLORS.blueMid} $bold>
+        {label}
+      </Subheading>
+    );
+  }, [listingData, interpretation]);
+
+  const Interest = useMemo(() => {
```
Comment (jinkang-0) [inline src/components/ListingDetails/index.tsx]: let's call this `interestSection` or something to stay consistent with the other naming, since this is technically not a component constructor

File: src/components/ListingDetails/index.tsx [hunk 36]
```diff
@@ -0,0 +1,421 @@
+/* eslint-disable react/jsx-no-comment-textnodes */
```
Comment (jinkang-0) [inline src/components/ListingDetails/index.tsx]: let's remove this, it doesn't seem to be necessary

File: src/components/ListingDetails/index.tsx (hunk 34 above)
Comment (ccatherinetan) [reply src/components/ListingDetails/index.tsx]: makes sense; changed!

**Review by jinkang-0 — APPROVED** [review]
lgtm!
